mungegithub/mungers/approval-handler.go

```diff
@@ -108,37 +106,28 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 
 		if len(fields) == 2 && strings.ToLower(fields[0]) == "/approve" && strings.ToLower(fields[1]) == "cancel" {
 			approverSet.Delete(*c.User.Login)
+			cancelSet.Insert(*c.User.Login)
 		}
 	}
+	glog.Infof("This is the cancel set %v", cancelSet)
 
+	//Checks that no valid approver has commented to cancel the approve label since last commit
+	//Checks that all files have an approver in the approverSet
 	for _, file := range files {
-		if !h.hasApproval(*file.Filename, approverSet, maxDepth) {
+		fileOwners := h.features.Repos.Assignees(*file.Filename)
+		if fileOwners.Intersection(cancelSet).Len() > 0 {
```

Comment:
Can you add the algorithm of cancelling into function comments too?
I think the comments doesn't reflect well enough what is being changed.